### PR TITLE
custom activiy to pass apikey and secet key

### PR DESCRIPTION
**Pull request by sndy35**
Please review the custom activity added to pass the asset details from app

### CODE REVIEW

File: ContentProtectionSampleApp/app/src/main/java/com/ooyala/sample/lists/ContentProtectionListActivity.java [hunk 1]
```diff
@@ -36,7 +36,19 @@ public void onCreate(Bundle savedInstanceState) {
     selectionMap = new LinkedHashMap<String, PlayerSelectionOption>();
     //Populate the embed map
 //    selectionMap.put("Adobe Pass Integration", new PlayerSelectionOption("VybW5lODrJ0uM9FBo7XTT6TNjTJfr_7G", "B3MDExOuTldXc1CiXbzAauYN7Iui", "http://www.ooyala.com", AdobePassSampleAppAcitivity.class) );
-    selectionMap.put("Ooyala Player Token", new PlayerSelectionOption("0yMjJ2ZDosUnthiqqIM3c8Eb8Ilx5r52", "c0cTkxOqALQviQIGAHWY5hP0q9gU", "http://www.ooyala.com", OoyalaPlayerTokenPlayerActivity.class) );
+   // selectionMap.put("Ooyala Player Token", new PlayerSelectionOption("0yMjJ2ZDosUnthiqqIM3c8Eb8Ilx5r52", "c0cTkxOqALQviQIGAHWY5hP0q9gU", "http://www.ooyala.com", OoyalaPlayerTokenPlayerActivity.class) );
+    selectionMap.put("OPT HLS", new PlayerSelectionOption("NjZ3B3ZDE6hN_sIoXotGDlI1s2cDLUwH", "BjcWYyOu1KK2DiKOkF41Z2k0X57l", "http://www.ooyala.com", OoyalaPlayerTokenPlayerActivity.class) );
```
Comment (brol1dev) [inline ContentProtectionSampleApp/app/src/main/java/com/ooyala/sample/lists/ContentProtectionListActivity.java]: are all of these cases part of your QA test cases?

File: ContentProtectionSampleApp/app/src/main/java/com/ooyala/sample/players/CustomActivity.java [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+        import  android.widget.Toast;
+
+        import com.ooyala.sample.R;
+
+        import java.util.ArrayList;
+        import java.util.HashMap;
+        import java.util.List;
+
+        /**
+ * This activity illustrates how we can dynamically add and play video using Skin SDK by adding the embed code and PCode.
```
Comment (brol1dev) [inline ContentProtectionSampleApp/app/src/main/java/com/ooyala/sample/players/CustomActivity.java]: This app doesn't use the Skin SDK, adding this comment can confuse customers. Remove the mention of SkinSDK please.

File: ContentProtectionSampleApp/app/src/main/java/com/ooyala/sample/players/CustomActivity.java [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+  * You can Select OoyalaDefault for BasicSkinPlayback also use it for playing Ooyala and VAST advertisements.
```
Comment (brol1dev) [inline ContentProtectionSampleApp/app/src/main/java/com/ooyala/sample/players/CustomActivity.java]: Don't understand this comment, where is BasicSkinPlayback? and what do you mean by OoyalaDefault.

Also there are no VAST ads in this app.

File: ContentProtectionSampleApp/app/src/main/java/com/ooyala/sample/players/CustomActivity.java [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+  * Also You can select Assets with GoogleIMA and Freewheel by selecting the respective players.
```
Comment (brol1dev) [inline ContentProtectionSampleApp/app/src/main/java/com/ooyala/sample/players/CustomActivity.java]: Remove this comment.

File: ContentProtectionSampleApp/app/src/main/java/com/ooyala/sample/players/CustomActivity.java [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+  *
+         *  * @author skumar
```
Comment (brol1dev) [inline ContentProtectionSampleApp/app/src/main/java/com/ooyala/sample/players/CustomActivity.java]: remove author

File: ContentProtectionSampleApp/app/src/main/java/com/ooyala/sample/players/CustomActivity.java [hunk 6]
```diff
@@ -0,0 +1,98 @@
+package com.ooyala.sample.players;
+
+        import android.app.Activity;
```
Comment (brol1dev) [inline ContentProtectionSampleApp/app/src/main/java/com/ooyala/sample/players/CustomActivity.java]: remove the extra space at the beginning of all of the imports.

File: ContentProtectionSampleApp/app/src/main/java/com/ooyala/sample/complete/MainActivity.java [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+
+import com.ooyala.sample.players.CustomActivity;
+
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+/**
+ * This is the opening activity for the app.
+ *
```
Comment (brol1dev) [inline ContentProtectionSampleApp/app/src/main/java/com/ooyala/sample/complete/MainActivity.java:23]: remove these extra lines in the comment if there are no comments.

File: ContentProtectionSampleApp/app/build.gradle [hunk 8]
```diff
@@ -1,5 +1,7 @@
+
```
Comment (brol1dev) [inline ContentProtectionSampleApp/app/build.gradle]: remove all the extra spaces added in this file

File: ContentProtectionSampleApp/app/src/main/java/com/ooyala/sample/players/CustomActivity.java [hunk 9]
```diff
@@ -0,0 +1,98 @@
...
+         *  * @author skumar
+  */
+        public class CustomActivity extends Activity {
+
+          public static String getName() {
+            return "Custom Activity";
+          }
+  private EditText embedCodeEditText;
+  private EditText pCodeEditText;
+  private EditText apikeyEditText;
```
Comment (brol1dev) [inline ContentProtectionSampleApp/app/src/main/java/com/ooyala/sample/players/CustomActivity.java]: use camel case for `apiKey` too

File: ContentProtectionSampleApp/app/src/main/java/com/ooyala/sample/players/CustomActivity.java [hunk 10]
```diff
@@ -0,0 +1,98 @@
...
+  private EditText secretEditText;
+  private EditText accountidEditText;
```
Comment (brol1dev) [inline ContentProtectionSampleApp/app/src/main/java/com/ooyala/sample/players/CustomActivity.java]: use camelCase for `accountId`

File: ContentProtectionSampleApp/app/src/main/java/com/ooyala/sample/players/OoyalaPlayerTokenPlayerActivity.java [hunk 11]
```diff
@@ -40,33 +45,44 @@
...
+  Boolean AUTOPLAY = false;
+  private String ACCOUNT_ID = "";
 
   /*
    * The API Key and Secret should not be saved inside your applciation (even in git!).
    * However, for debugging you can use them to locally generate Ooyala Player Tokens.
    */
-  private final String APIKEY = "Use this for testing, don't keep your secret in the application";
-  private final String SECRET = "Use this for testing, don't keep your secret in the application";
+  private String APIKEY = "BjcWYyOu1KK2DiKOkF41Z2k0X57l.0I-V4";
```
Comment (brol1dev) [inline ContentProtectionSampleApp/app/src/main/java/com/ooyala/sample/players/OoyalaPlayerTokenPlayerActivity.java]: remove api key and secret values, these are supposed to be secret. Are this from a test account? If not you'll need to completely remove them from GIT.

File: ContentProtectionSampleApp/app/src/main/java/com/ooyala/sample/players/OoyalaPlayerTokenPlayerActivity.java [hunk 12]
```diff
@@ -78,7 +94,8 @@ public void onCreate(Bundle savedInstanceState) {
     player.addObserver(this);
 
     if (player.setEmbedCode(EMBED)) {
-      player.play();
+      if(AUTOPLAY)
```
Comment (brol1dev) [inline ContentProtectionSampleApp/app/src/main/java/com/ooyala/sample/players/OoyalaPlayerTokenPlayerActivity.java]: add extra braces for this if, it doesn't matter if it's only one line.

File: ContentProtectionSampleApp/app/src/main/res/values/strings.xml [hunk 13]
```diff
@@ -3,5 +3,10 @@
 
     <string name="hello">Hello World, ContentProtectionSampleApp!</string>
     <string name="app_name">ContentProtectionSampleApp</string>
+       <string name="open">Open</string>
+        <string name="pcode">pcode:</string>
+        <string name="embed">embed:</string>
+        <string name="select_player">selectPlayer:</string>
```
Comment (brol1dev) [inline ContentProtectionSampleApp/app/src/main/res/values/strings.xml]: I don't see you use this value at all. You can remove it.

File: ContentProtectionSampleApp/app/src/main/res/values/strings.xml [hunk 14]
```diff
@@ -3,5 +3,10 @@
 
     <string name="hello">Hello World, ContentProtectionSampleApp!</string>
     <string name="app_name">ContentProtectionSampleApp</string>
+       <string name="open">Open</string>
```
Comment (brol1dev) [inline ContentProtectionSampleApp/app/src/main/res/values/strings.xml]: I don't see you use this value at all. You can remove it.

**Review by brol1dev — CHANGES_REQUESTED** [review]

**Comment by sndy35** [comment]
Implemented the review comments. Please review

File: ContentProtectionSampleApp/app/src/main/java/com/ooyala/sample/lists/ContentProtectionListActivity.java [hunk 15]
```diff
@@ -36,7 +36,7 @@ public void onCreate(Bundle savedInstanceState) {
     selectionMap = new LinkedHashMap<String, PlayerSelectionOption>();
     //Populate the embed map
 //    selectionMap.put("Adobe Pass Integration", new PlayerSelectionOption("VybW5lODrJ0uM9FBo7XTT6TNjTJfr_7G", "B3MDExOuTldXc1CiXbzAauYN7Iui", "http://www.ooyala.com", AdobePassSampleAppAcitivity.class) );
-    selectionMap.put("Ooyala Player Token", new PlayerSelectionOption("0yMjJ2ZDosUnthiqqIM3c8Eb8Ilx5r52", "c0cTkxOqALQviQIGAHWY5hP0q9gU", "http://www.ooyala.com", OoyalaPlayerTokenPlayerActivity.class) );
+   // selectionMap.put("Ooyala Player Token", new PlayerSelectionOption("0yMjJ2ZDosUnthiqqIM3c8Eb8Ilx5r52", "c0cTkxOqALQviQIGAHWY5hP0q9gU", "http://www.ooyala.com", OoyalaPlayerTokenPlayerActivity.class) );
```
Comment (brol1dev) [inline ContentProtectionSampleApp/app/src/main/java/com/ooyala/sample/lists/ContentProtectionListActivity.java]: why did you comment this asset? This is the only one we have in here. Please add it back.

File: ContentProtectionSampleApp/app/src/main/java/com/ooyala/sample/players/CustomActivity.java [hunk 16]
```diff
@@ -0,0 +1,88 @@
...
+    private String secret;
+    private String accountId;
+    private CheckBox autoPlayCheckBox;
+    private boolean autoPlay;
+
+  @Override
+    public void onCreate(Bundle savedInstanceState) {
+      super.onCreate(savedInstanceState);
+      setContentView(R.layout.embed_pcode_layout);
+      embedCodeEditText = (EditText) findViewById(R.id.embed_edit_text);
```
Comment (brol1dev) [inline ContentProtectionSampleApp/app/src/main/java/com/ooyala/sample/players/CustomActivity.java]: It is now redundant to cast the type of the object when using `findViewById` you can remove the casting type now

File: ContentProtectionSampleApp/app/src/main/java/com/ooyala/sample/players/CustomActivity.java [hunk 17]
```diff
@@ -0,0 +1,88 @@
...
+import  android.widget.Toast;
+import com.ooyala.sample.R;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+
+
+public class CustomActivity extends Activity {
+    public static String getName() {
+        return "Custom Activity";
```
Comment (brol1dev) [inline ContentProtectionSampleApp/app/src/main/java/com/ooyala/sample/players/CustomActivity.java]: Use another name here. My suggestion is: `Asset Options`

File: ContentProtectionSampleApp/app/build.gradle (hunk 8 above)
Comment (brol1dev) [reply ContentProtectionSampleApp/app/build.gradle]: there's still extra space here.

**Review by brol1dev — CHANGES_REQUESTED** [review]
Added more comments. That should be all that is remaining.

**Review by brol1dev — APPROVED** [review]
LGTM

**Comment by brol1dev** [comment]
@akshaydch11 can you review this one too? You can merge if you're ok with it.

**Pull request merged**